drivers/counter/counter_esp32_rtc.c

```diff
@@ -63,12 +63,12 @@ static int counter_esp32_init(const struct device *dev)
 			       (clock_control_subsys_t)ESP32_CLOCK_CONTROL_SUBSYS_RTC_SLOW,
 			       &data->clk_src_freq);
 
-	int ret = esp_intr_alloc(cfg->irq_source,
-				ESP_PRIO_TO_FLAGS(cfg->irq_priority) |
-				ESP_INT_FLAGS_CHECK(cfg->irq_flags),
-				(ESP32_COUNTER_RTC_ISR_HANDLER)counter_esp32_isr,
-				(void *)dev,
-				NULL);
+	int ret =
+		esp_intr_alloc(cfg->irq_source,
+			       ESP_PRIO_TO_FLAGS(cfg->irq_priority) |
+				       ESP_INT_FLAGS_CHECK(cfg->irq_flags | ESP_INTR_FLAG_SHARED),
+			       (ESP32_COUNTER_RTC_ISR_HANDLER)counter_esp32_isr, (void *)dev, NULL);
+
```

Comment:
that's some interesting indentation story going, how about adding an intermediate variable or two? Maybe just one for `ESP_PRIO_TO_FLAGS(cfg->irq_priority) | ESP_INT_FLAGS_CHECK(cfg->irq_flags | ESP_INTR_FLAG_SHARED)`